frontend/pages/admin/AppSettingsPage/AppSettingsPage.tsx

```diff
@@ -63,68 +98,174 @@ const AppSettingsPage = (): JSX.Element => {
-    const yOffset = -215; // headers and tabs
-    const element = document.getElementById(elementId);
+  const showSection = (linkString: string) => {
+    setShowInfo(false);
+    setShowWebAddress(false);
+    setShowSso(false);
+    setShowSmtp(false);
+    setShowAgents(false);
+    setShowHostStatusWebhook(false);
+    setShowStatistics(false);
+    setShowAdvanced(false);
 
-    if (element) {
-      const top =
-        element.getBoundingClientRect().top + window.pageYOffset + yOffset;
-      window.scrollTo({ top });
+    switch (linkString) {
+      case "webaddress":
+        setShowWebAddress(true);
+        return;
+      case "sso":
+        setShowSso(true);
+        return;
+      case "smtp":
+        setShowSmtp(true);
+        return;
+      case "agents":
+        setShowAgents(true);
+        return;
+      case "host-status-webhook":
+        setShowHostStatusWebhook(true);
+        return;
+      case "statistics":
+        setShowStatistics(true);
+        return;
+      case "advanced":
+        setShowAdvanced(true);
+        return;
+      default:
+        setShowInfo(true);
     }
   };
 
+  useEffect(() => {
+    showSection(sectionTitle);
+  }, [sectionTitle]);
+
+  const renderSection = () => {
+    if (!isLoadingConfig && appConfig) {
+      return (
+        <>
+          {showInfo && (
+            <Info appConfig={appConfig} handleSubmit={onFormSubmit} />
+          )}
+          {showWebAddress && (
+            <WebAddress appConfig={appConfig} handleSubmit={onFormSubmit} />
+          )}
+          {showSso && <Sso appConfig={appConfig} handleSubmit={onFormSubmit} />}
+          {showSmtp && (
+            <Smtp appConfig={appConfig} handleSubmit={onFormSubmit} />
+          )}
+          {showAgents && (
+            <AgentOptions appConfig={appConfig} handleSubmit={onFormSubmit} />
+          )}
+          {showHostStatusWebhook && (
+            <HostStatusWebhook
+              appConfig={appConfig}
+              handleSubmit={onFormSubmit}
+            />
+          )}
+          {showStatistics && (
+            <Statistics appConfig={appConfig} handleSubmit={onFormSubmit} />
+          )}
+          {showAdvanced && (
+            <Advanced appConfig={appConfig} handleSubmit={onFormSubmit} />
+          )}
+        </>
+      );
+    }
+
+    return <></>;
+  };
+
   return (
     <div className={`${baseClass} body-wrap`}>
       <p className={`${baseClass}__page-description`}>
-        Set your organization information and configure SAML and SMTP.
+        Set your organization information and configure SSO and SMTP
       </p>
       <div className={`${baseClass}__settings-form`}>
         <nav>
           <ul className={`${baseClass}__form-nav-list`}>
             <li>
-              <a onClick={() => scrollInto("organization-info")}>
+              <Link
+                className={`${baseClass}__nav-link${
+                  showInfo ? " active-nav" : ""
```

Comment:
Then you'd change this to be `activeSection === "info" ? " active-nav" : ""`

If you wanted a cleaner approach to this you'd need to use a config object that you iterate through and generate the list of sections dynamically. You could use the same config object to generate the section elements, as well, depending on how far you wanted to take it. For example, something like this could be used to define the data only once, then use loops to generate the rest of the component based on this array: 

```javascript
  const settingsSections = [
    {
      title: "Organization info",
      key: "info",
      state: "showInfo",
      element: "Info",
      active: true,
    },
    {
      title: "Organization info",
      key: "webaddress",
      state: "showWebAddress",
      element: "WebAddress",
      active: false,
    },
  ];
```

While that would result in less lines of code, it would also make the component a bit more complicated to maintain. I think it would be worthwhile if the settings page was very dynamic and changed a lot, or had different options for different users. Since it's relatively static and we aren't adding sections very often, I think your approach of hardcoding everything is fine for the time being.